go-selinux/label/label_selinux.go

```diff
@@ -118,7 +119,19 @@ func SocketLabel() (string, error) {
 // SetKeyLabel takes a process label and tells the kernel to assign the
 // label to the next kernel keyring that gets created
 func SetKeyLabel(processLabel string) error {
-	return selinux.SetKeyLabel(processLabel)
+	if processLabel == "" && selinux.GetEnabled() {
+		if _, err := KeyLabel(); err == io.EOF {
+			// If we got io.EOF err, this means it is the first time write "" to keycreate
+			// So we can ignore it
+			return nil
+		}
+	}
+	err := selinux.SetKeyLabel(processLabel)
+	if err != nil && processLabel == "" && os.IsNotExist(err) {
```

Comment:
I don't think `processLabel == ""` should be here. really it should just be:

```
if os.IsNotExist(err) {
}
```

With the note that `os.IsNotExist(nil) == false`.

Reply:
Thanks for your review. I have update it.